### PR TITLE
Added and used flag to display filings filed by Registry Staff

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#6814

*Description of changes:*
- displayed Filed By according to staff flag or certified by value, else fall back to Unknown
- saved isStaff flag in AR, Correction, COD and COA
- updated app version

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Dashboard/FilingHistoryList.vue [hunk 1]
```diff
@@ -922,7 +922,7 @@ export default {
           title: this.filingTypeToName(filingType),
           subtitle,
           filingId: header.filingId,
-          filingAuthor: 'Registry Staff', // TBD
+          filingAuthor: 'Registry Staff',
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:925]: As far as I know, alteration filings are filed by Registry Staff (Ops) only.

File: src/components/Dashboard/FilingHistoryList.vue [hunk 2]
```diff
@@ -541,7 +541,7 @@ export default {
           filingType,
           title: this.filingTypeToName(filingType, agmYear),
           filingId: header.filingId,
-          filingAuthor: header.certifiedBy,
+          filingAuthor: header.isStaff ? 'Registry Staff' : (header.certifiedBy || 'Unknown'),
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:544]: I wasn't sure if this was worth a new function to call from several places... so I didn't. Thoughts?

File: src/components/Dashboard/FilingHistoryList.vue [hunk 3]
```diff
@@ -541,7 +541,7 @@ export default {
           filingType,
           title: this.filingTypeToName(filingType, agmYear),
           filingId: header.filingId,
-          filingAuthor: header.certifiedBy,
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:544]: Note that we were missing the Registry Staff fallback for some of our filings. (Oops.) Fixed now.

Note also that the new behaviour is the same as the previous behaviour (except for the fixed filing types), and is backward compatible with current filings (that don't have the `isStaff` header property).

File: src/components/Dashboard/FilingHistoryList.vue (hunk 2 above)
Comment (cameron-freshworks) [reply src/components/Dashboard/FilingHistoryList.vue:544]: Oh ya know i'm fine with ternary!

File: src/components/Dashboard/FilingHistoryList.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/Dashboard/FilingHistoryList.vue:925]: Wait, is this true? 
Doesn't the client themselves file the Alterations? The staff only load the business and create the draft right?

Comment (severinbeauvais) [reply src/components/Dashboard/FilingHistoryList.vue:925]: I'll review this tomorrow!

File: src/components/Dashboard/FilingHistoryList.vue (hunk 2 above)
Comment (andrepestana-aot) [reply src/components/Dashboard/FilingHistoryList.vue:544]: I'm ok with that too.

File: src/components/Dashboard/FilingHistoryList.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/Dashboard/FilingHistoryList.vue:925]: You're right: alterations can be filed by users. And that code is correct (on line 684)...

Line 925 (above) is actually for the Transition Filings. I typed in the wrong filing type in my comment :P

**Comment by severinbeauvais** [comment]
I can't see to tell SonarCloud to ignore the code duplication, but I'm not going to fix it as that's a refactoring exercise not in this ticket's scope. So please ignore that specific check.

File: src/components/Dashboard/FilingHistoryList.vue [hunk 4]
```diff
@@ -1003,6 +1003,12 @@ export default {
       }
     },
 
+    filingAuthor (header: HeaderIF): string {
+      if (header.isStaff) return 'Registry Staff'
+      if (header.certifiedBy) return header.certifiedBy
+      return 'Unknown'
+    },
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:1010]: I know y'all loved the ternary (with a || thrown in) but I think THIS code is easier to understand 😁

File: src/interfaces/header-interface.ts [hunk 5]
```diff
@@ -2,11 +2,12 @@ import { FilingTypes } from '@/enums'
...
-  status: string;
-  availableInColinOnly?: boolean;
-  availableOnPaperOnly?: boolean;
+  date: string
+  filingId: number
+  name: FilingTypes
+  status: string
+  availableInColinOnly?: boolean
+  availableOnPaperOnly?: boolean
+  isStaff?: boolean
```
Comment (severinbeauvais) [inline src/interfaces/header-interface.ts:11]: Optional since older filings don't have this field.

File: tests/unit/FilingHistoryList.spec.ts [hunk 6]
```diff
@@ -1315,7 +1316,8 @@ describe('Filing History List - alterations', () => {
             name: 'alteration',
             date: '2020-03-24T19:20:05.670859+00:00',
             status: 'COMPLETED',
-            filingId: 85114
+            filingId: 85116,
```
Comment (severinbeauvais) [inline tests/unit/FilingHistoryList.spec.ts:1319]: This is the dumbest thing: SonarCloud was complaining about the duplicated 2 lines, so I changed the first line to work around that error 😛

File: src/components/Dashboard/FilingHistoryList.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/Dashboard/FilingHistoryList.vue:925]: Sounds good! You can resolve when ready 👍

**Comment by severinbeauvais** [comment]
There's a new design in the works. I am closing this PR and will open another (cleaner) one.

**Pull request closed without merging**